src/streamlit_pydantic/ui_renderer.py

```diff
@@ -143,11 +143,21 @@ def render_ui(self) -> Dict:
                 # Set property key as fallback title
                 property["title"] = _name_to_title(property_key)
 
+            # check if the input_class is an instance
+            if isinstance(self._input_class, BaseModel):
+                instance_value = self._input_class.dict().get(property_key)
```

Comment:
nit: I'm not sure about the performance efficiency of calling `.dict()`. It might be better to have this outside of the property loop. We can probably just construct those two dicts above the loop, e.g.:

```python
if isinstance(self._input_class, BaseModel):
    instance_dict = self._input_class.dict()
    instance_dict_by_alias = self._input_class.dict()

for property_key in self._schema_properties.keys():
    ...
    if instance_dict:
        instance_dict.get(property_key)
    ....
 ```

Reply:
Thanks thats a good suggestion which I've now implemented.

There are some concerns about performance generally but I'll start a separate conversation about it below.